src/Core/Datatypes/DyadicTensor.h

```diff
@@ -0,0 +1,306 @@
+        eigvecs_ = other.getEigenvectors();
+        eigvals_ = other.getEigenvalues();
+        haveEigens_ = true;
+        reorderTensorValues();
+      }
+
+      DyadicTensorGeneric(const Eigen::TensorFixedSize<Number, Eigen::Sizes<Dim0, Dim1>>& other)
+          : parent()
+      {
+        for (size_t i = 0; i < Dim0; ++i)
+          for (size_t j = 0; j < Dim1; ++j)
+            (*this)(index(i), index(j)) = other(index(i), index(j));
+      }
+
+      DyadicTensorGeneric(const DenseMatrixGeneric<Number>& mat)
+      {
+        if (mat.nrows() != Dim0)
+          THROW_INVALID_ARGUMENT("The number of rows of the input matrix must be " + Dim0);
+        if (mat.ncols() != Dim1)
+          THROW_INVALID_ARGUMENT("The length of rows of the input matrix must be " + Dim1);
+        for (size_t i = 0; i < Dim0; ++i)
+          for (size_t j = 0; j < Dim1; ++j)
+            (*this)(index(i), index(j)) = mat(index(i), index(j));
+      }
+
+      using parent::operator=;
+      using parent::contract;
+
+      template <typename OtherDerived>
+      bool operator!=(const OtherDerived& other)
+      {
+        return !operator==(other);
+      }
+
+      template <typename OtherDerived>
+      bool operator==(const OtherDerived& other)
+      {
+        auto otherTensor = static_cast<DyadicTensorGeneric<Number, Dim0, Dim1>>(other);
+        if (Dim0 != otherTensor.dimension(0) || Dim1 != otherTensor.dimension(1)) return false;
+        for (size_t i = 0; i < Dim0; ++i)
+          for (size_t j = 0; j < Dim1; ++j)
+            if ((*this)(index(i), index(j)) != otherTensor(index(i), index(j))) return false;
+        return true;
+      }
+
+      void setEigenVectors(const std::vector<DenseColumnMatrixGeneric<Number>>& eigvecs)
+      {
+        eigvecs_ = eigvecs;
+        setEigenvaluesFromEigenvectors();
+        normalizeEigenvectors();
+        haveEigens_ = true;
+        setTensorValues();
+        reorderTensorValues();
+      }
+
+      // This function is listed as something that will be added to Eigen::Tensor in the future.
+      // Usage of this function should be replaced with Eigen's asMatrix function when it is
+      // implemented.
+      Eigen::Matrix<Number, Dim0, Dim1> asMatrix()
+      {
+        Eigen::Matrix<Number, Dim0, Dim1> mat;
+        for (size_t i = 0; i < Dim0; ++i)
+          for (size_t j = 0; j < Dim1; ++j)
+            mat(index(i), index(j)) = (*this)(index(i), index(j));
+        return mat;
+      }
+
+      DenseColumnMatrixGeneric<Number> getEigenvector(int index)
+      {
+        if (!haveEigens_) buildEigens();
+        return eigvecs_[index];
+      }
+
+      std::vector<DenseColumnMatrixGeneric<Number>> getEigenvectors()
+      {
+        if (!haveEigens_) buildEigens();
+        return eigvecs_;
+      }
+
+      std::vector<Number> getEigenvalues()
+      {
+        if (!haveEigens_) buildEigens();
+        return eigvals_;
+      }
+
+      SCISHARE friend std::ostream& operator<<(
```

Comment:
I'm thinking these need to be declared as friends outside of the class to make Windows/Mac happy. An inline friend declaration usually isn't enough.